### PR TITLE
Better exception message for when default value is an enum

### PR DESCRIPTION
**Pull request by forrest79**
This solves an issue #1376.

### CODE REVIEW

File: test/unit/NodeCompiler/CompileNodeToValueTest.php [hunk 1]
```diff
@@ -741,6 +741,29 @@ class Bat {
...
+        }
+        PHP;
+
+        $reflector = new DefaultReflector(new AggregateSourceLocator([
+            new StringSourceLocator($phpCode, $this->astLocator),
+            new PhpInternalSourceLocator($this->astLocator, $this->sourceStubber),
+        ]));
+        $classInfo = $reflector->reflectClass('Bat');
+
+        $this->expectException(UnableToCompileNode::class);
```
Comment (Ocramius) [inline test/unit/NodeCompiler/CompileNodeToValueTest.php]: We should check at least part of the exception message in this test: it seems like the mutation test reports no behavioral change (removes `throw` and is happy with that), which indicates that an assertion is to be added here

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by forrest79** [comment]
Sorry for a delay, I was working remotely for a couple of days. I added a message check. You can look at it.

File: test/unit/NodeCompiler/CompileNodeToValueTest.php [hunk 2]
```diff
@@ -760,8 +761,13 @@ class Bat {
         ]));
         $classInfo = $reflector->reflectClass('Bat');
 
-        $this->expectException(UnableToCompileNode::class);
-        $classInfo->getConstant('ONE_VALUE')->getValue();
+        try {
+            $classInfo->getConstant('ONE_VALUE')->getValue();
+        } catch (Throwable $e) {
+            self::assertInstanceOf(UnableToCompileNode::class, $e);
+            self::assertStringContainsString('An enum expression Foo::ONE is not supported in class Bat in file ', $e->getMessage());
```
Comment (Ocramius) [inline test/unit/NodeCompiler/CompileNodeToValueTest.php]: Can use `expectExceptionMessage`, rather than adding a try-catch 😁

**Review by Ocramius — CHANGES_REQUESTED** [review]
CS issues fixable with some minimal assertion improvements (by removing try-catch)

File: test/unit/NodeCompiler/CompileNodeToValueTest.php (hunk 2 above)
Comment (forrest79) [reply test/unit/NodeCompiler/CompileNodeToValueTest.php]: Done

**Comment by Ocramius** [comment]
I think the failures are unrelated to your patch, and instead depend on https://github.com/shivammathur/setup-php/issues/799

**Review by Ocramius — APPROVED** [review]
LGTM, thanks @forrest79!

I will re-run tests and merge once the upstream issue with the CI pipeline is resolved, then releasing :muscle:

**Comment by Ocramius** [comment]
Needs https://github.com/shivammathur/php-ubuntu/pull/3

**Review by Ocramius — APPROVED** [review]
Excellent, thanks @forrest79!

**Pull request merged**